torch_xla/core/xla_model.py

```diff
@@ -1012,7 +1012,15 @@ def convert_fn(tensors):
     devices = [str(device)] * len(tensors)
     shardings = None
     if input_sharding:
-      shardings = [input_sharding.xla_spec(t) for t in tensors]
+      if isinstance(input_sharding, list):
+        shardings = [None] * len(tensors)
+        for i, tensor in enumerate(tensors):
+          for sharding in input_sharding:
+            if sharding.can_apply(tensor):
+              shardings[i] = sharding.xla_spec(tensor)
+              break
```

Comment:
Maybe we also add some comments about the first match is applied?

Reply:
Done, thanks Yeounoh!

Reply:
Is first match something common practice? Or will it be better:
1. either one sharding spec for everything;
2. or one sharding spec for each input and in order.

That the behavior is well defined. Otherwise, this "compatible check" is totally a black box for the user.

Reply:
It's definitely not ideal... Probably better would be to have the user provide sharding specs in a structure matching that of the inputs and avoid the black-box compatibility check, e.g.:

```python
# Inputs of the form:
{"a": torch.randn(16), "b": torch.randn(16, 16), "c": torch.randn(16, 16, 16)}

# Then sharding_spec input would be:
{"a": ShardingSpec(mesh, (0,)), "b": ShardingSpec(mesh, (0, None)), "c": ShardingSpec(mesh, (0, None, None))}
```

This is a bit of a refactor of the TensorToXlaArena code though, so I took the easy way out. I'll go ahead and take a stab at the cleaner approach now.

Reply:
No worries. Take your time.